frontend/test/resolution/testFieldAccess.cpp

```diff
@@ -89,10 +89,52 @@ static void test2() {
   assert(field->id() == results->resolutionById().byAst(ret).toId());
 }
 
+static void test3() {
+  printf("test3\n");
+  Context ctx;
+  Context* context = &ctx;
+  ErrorGuard guard(context);
+
+  auto path = UniqueString::get(context, "test3.chpl");
+  std::string contents = R""""(
+    class Outer {
+        var outerField: int;
+        record Inner {
+            proc innerMethod() {
+                return outerField;
+            }
+        }
+
+        var inner: Inner;
+        proc outerMethod() {
+            return inner.innerMethod();
+        }
+    }
+   )"""";
+  setFileText(context, path, contents);
+
+  const ModuleVec& vec = parseToplevel(context, path);
+  assert(vec.size() == 1);
+
+  const Module* m = vec[0]->toModule();
+  assert(m);
+  assert(m->numStmts() == 1);
+
+  const AggregateDecl* outerAd = m->stmt(0)->toAggregateDecl();
+  const AggregateDecl* innerId = outerAd->declOrComment(1)->toAggregateDecl();
+  const Function* innerMethod = innerId->declOrComment(0)->toFunction();
+
+  scopeResolveFunction(context, innerMethod->id());
+  assert(guard.numErrors() == 1);
+  auto firstError = guard.error(0).get();
+  assert(firstError->type() == ErrorType::NestedClassFieldRef);
```

Comment:
I never realized you could do this with `->type()` and that will make my life so much easier...